libs/geography/src/reducers/country-entities/country-entities.reducer.ts

```diff
@@ -7,15 +7,18 @@ import { daffCountryEntitiesInitialState } from './country-entities-initial-stat
 /**
  * Reducer function that catches actions and changes/overwrites country entities state.
  */
-export function daffCountryEntitiesReducer<T extends DaffCountry>(
+export function daffCountryEntitiesReducer<T extends DaffCountry = DaffCountry>(
   state = daffCountryEntitiesInitialState,
   action: DaffGeographyActions<T>
 ): DaffCountryEntityState<T> {
   const adapter = getCountryAdapter<T>();
 
   switch (action.type) {
     case DaffGeographyActionTypes.CountryLoadSuccessAction:
-      return adapter.upsertOne(action.payload, state);
+      return adapter.upsertOne({
+        ...action.payload,
+        loaded: true
+      }, state);
 
     case DaffGeographyActionTypes.CountryListSuccessAction:
       return adapter.upsertMany(action.payload, state);
```

Comment:
I think it might be a little more accurate to do something like:
```
case DaffGeographyActionTypes.CountryListSuccessAction:
      return adapter.upsertMany(action.payload.map(country => (
          { ...country, loaded: false}
      )), state);
```
Just so the `loaded` field isn't true or null. It'll be true or false.

Reply:
changed

Reply:
actually I'm not sure if this is a good idea. If the country has already been fully loaded, then a list success would overwrite this. I think it should be modified to
```ts
return adapter.upsertMany(action.payload.map(country => (
          { ...country, loaded: country.loaded || false}
      )), state);
```

LMK what you think

Reply:
Actually the problem is broader, any listed countries will overwrite the subdivisions of already loaded countries. I'm going to rework this reducer case.

Reply:
Okay I just pushed a new implementation that doesn't overwrite fully loaded countries.